backend/apps/notifications/services.py

```diff
@@ -0,0 +1,39 @@
+from typing import List
+from django.db import transaction
+from apps.notifications.models import Notification
+from apps.users.models import BaseUser
+
+@transaction.atomic
+def notification_create(
+    *,
+    recipient: BaseUser,
+    message: str,
+    notification_type: str,
+    is_read: bool = False,
+) -> Notification:
+    """
+    Create a new notification for a user.
+    """
+    notification = Notification.objects.create(
+        recipient=recipient,
+        message=message,
+        notification_type=notification_type,
+        is_read=is_read,
+    )
+    return notification
+
+@transaction.atomic
+def notification_mark_as_read(notification: Notification) -> Notification:
```

Comment:
This function should instead be notification update like the other functions, and use model_update like the others.

Or, keep this, add notification_update and call notification_update from this func with the read parameter.